src/codegen/compile.ml

```diff
@@ -11205,107 +11007,122 @@ and compile_prim_invocation (env : E.t) ae p es at =
     IC.canister_version env
 
   | OtherPrim "crc32Hash", [e] ->
-    SR.UnboxedWord32 Type.Nat32,
+    SR.Vanilla,
```

Comment:
```suggestion
    SR.UnboxedWord64 Type.Nat32
```
Otherwise this will (with rtti) be missing the tag bits required by a vanilla value, I think

Reply:
Indeed, a bug. I fixed it and added a test case.